src/main/java/org/apache/maven/shared/incremental/IncrementalBuildHelper.java

```diff
@@ -323,25 +335,23 @@ public String[] beforeRebuildExecution( IncrementalBuildHelperRequest incrementa
     public void afterRebuildExecution( IncrementalBuildHelperRequest incrementalBuildHelperRequest )
         throws MojoExecutionException
     {
-        DirectoryScanner diffScanner = getDirectoryScanner();
-        // now scan the same directory again and create a diff
-        diffScanner.scan();
-        DirectoryScanResult scanResult = diffScanner.diffIncludedFiles( filesBeforeAction );
-
         File mojoConfigBase = getMojoStatusDirectory();
-        File mojoConfigFile = new File( mojoConfigBase, CREATED_FILES_LST_FILENAME );
 
-        try
-        {
-            FileUtils.fileWriteArray( mojoConfigFile, scanResult.getFilesAdded() );
-        }
-        catch ( IOException e )
-        {
-            throw new MojoExecutionException( "Error while storing the mojo status", e );
-        }
+        writeChangedFiles(
+                getDirectoryScanner(),
```

Comment:
no need to call a getter method in the same class, just use the directoryScanner field directly